src/gwt/src/org/rstudio/studio/client/workbench/views/environment/EnvironmentPresenter.java

```diff
@@ -474,17 +480,30 @@ public void onError(ServerError error)
 
    void onImportDatasetFromCSV()
    {
-      view_.bringToFront();
-      DataImportDialog dataImportDialog = new DataImportDialog("Import CSV", new OperationWithInput<String>()
-      {
-         @Override
-         public void execute(String input)
-         {
-            
-         }
+      final String action = "Preparing data import";
+      dependencyManager_.withDataImport(
+         action, 
+         new Command() {
+            @Override
+            public void execute()
+            {
+               view_.bringToFront();
+               DataImportDialog dataImportDialog = new DataImportDialog(
+                     "Data Import",
+                     new OperationWithInput<DataImportOptions>()
+               {
+                  @Override
+                  public void execute(final DataImportOptions options)
+                  {
+                     final DataImportScript dataImportScript = new DataImportScript();
+                     String code = dataImportScript.getImportScript(options);
+                     eventBus_.fireEvent(new SendToConsoleEvent(code, true)); 
```

Comment:
You might also want to add `true` as the third argument to also set focus to the console (this is debatable though so whatever you think best is fine)